src/devices/Common/System/Device/DelayHelper.cs

```diff
@@ -45,10 +54,24 @@ public static void DelayMicroseconds(int microseconds, bool allowThreadYield)
                 {
                     spinWait.SpinOnce();
                 }
-                while ((ulong)(Stopwatch.GetTimestamp() - start) < minimumTicks);
+                while (Stopwatch.GetTimestamp() < target);
```

Comment:
I think the previous logic here was preventing overflows (you'd get double overflow so would just work)

Reply:
I don't think there is a real need to account for overflow here, it just makes the code harder to read at little benefit (we don't account for this in `System.Diagnostics.Stopwatch` either, for similar reasons).
Even if `Frequency` is 1ns ticks, it would take 292.3 years to overflow, in which case the fix is for the user to install updates and reboot their computer 😄

Windows currently returns 100ns ticks (but supports more precise), Unix/POSIX returns 1ns ticks (and is hardcoded to do so). Hardware has a latency of ~300ns to even query the timer.

Reply:
fair enough 😄